File: Programming Basics/JavaScript/While loops/Labs/graduation.js
function solve(input) {
    const studentName = input[0];

    let grade = 0;
    let fails = 0;
    let sum = 0;
    let index = 1;

    while (grade < 12) {
        const marks = Number(input[index]);
        if (marks < 4) {
            if (fails == 1) break;
            fails++;
        } else {
            sum += marks;
            grade++;
            fails = 0;
        }

        index++;
    }

    if (grade == 12) console.log(`${studentName} graduated. Average grade: ${(sum / 12).toFixed(2)}`);
    else console.log(`${studentName} has been excluded at ${grade + 1} grade`);
}

File: Programming Basics/C++/While loops/Labs/Graduation/Graduation.cpp
#include <iostream>
using namespace std;

int main()
{
    string studentName;
    cin >> studentName;

    int grade = 0;
    int fails = 0;
    double sum = 0;

    while (grade < 12)
    {
        double marks;
        cin >> marks;

        if (marks < 4)
        {
            if (fails == 1) break;
            fails++;
        }
        else
        {
            sum += marks;
            grade++;
            fails = 0;
        }
    }

    if (grade == 12)
    {
        double average = sum / 12;

        cout.setf(ios::fixed);
        cout.precision(2);
        cout << studentName << " graduated. Average grade: " << average << endl;
    }
    else
    {
        cout << studentName << " has been excluded at " << grade + 1 << " grade";
    }
}
